Add the load user actions. (#29)
plantanapp/documentation: docs/Actions/load-user.md

---
id: LoadUser
title: Load User
sidebar_label: Load User
---

An action that allows altering the user which the subsequent actions refer to. The user change affects all actions after the `Load User` action and after all actions have finished executing the user will be reverted back to the initial one (i.e. the logged-in user). By default, all actions that need a user for their execution will refer to the currently logged in user.

This action also supports loading multiple users by providing multiple user identifiers. The last user identifier provided will also be used as the context user under which all actions execute. Loading multiple users creates a hidden context entity that is leveraged by the following actions which support this:

- [Grant Role](/docs/Actions/grant-role.md)
- [Revoke Role](/docs/Actions/revoke-role.md)
- [Authorize User](/docs/Actions/authorize-user.md)
- [Unauthorize User](/docs/Actions/unauthorize-user.md)
- [Delete User](/docs/Actions/delete-user.md)
- [Send Email](/doc/Actions/send-email.md).

> When used in conjunction with actions such as [Grant Role](/docs/Actions/grant-role.md), the `Grant Role` action will grant the role specified to all users loaded by this action.

Reviewer: ```suggestion
When used in conjunction with the actions above, the logic applies to all users that have been loaded into the context by the `Load User` action. For example, the `Grant Role` action will assign the specified role to each and every user loaded into the context.
```

> 

Reviewer: ```suggestion

Calling the `Load User` action the second time inside the same workflow will replace the user or the list of users in the context. It never appends to the list that is already loaded.

```

Author: This is not the actual behavior, I committed this change with the proper explanation.

This action also supports loading multiple users by providing multiple user identifiers. The last user identifier provided will also be used as the context user under which all actions execute. Loading multiple users creates a hidden context entity that is leveraged by the following actions which support this:

- [Grant Role](/docs/Actions/grant-role.md)
- [Revoke Role](/docs/Actions/revoke-role.md)
- [Authorize User](/docs/Actions/authorize-user.md)
- [Unauthorize User](/docs/Actions/unauthorize-user.md)
- [Delete User](/docs/Actions/delete-user.md)
- [Send Email](/doc/Actions/send-email.md).

When used in conjunction with the actions above, the logic applies to all users that have been loaded into the context by the `Load User` action. For example, the `Grant Role` action will assign the specified role to each and every user loaded into the context.

> Calling the `Load User` action the second time inside the same workflow will replace the context user that the actions use and add to the list of context users(the Users entity) that actions such as `Grant Role` leverage.

Reviewer: ```suggestion
Calling the `Load User` action the second time inside the same workflow will replace the context user that the actions use and add to the list of context users (the Users entity) that actions such as `Grant Role` leverage.
```

> Calling the `Load User` action the second time inside the same workflow will replace the context user that the actions use and add to the list of context users (the Users entity) that actions such as `Grant Role` leverage. If a user is already present in the Users entity he will not be added again.

Reviewer: Avoid "he" or "she" as it discriminates. In this case "it" works. Or, prefer plural, as "they" is gender-neutral.

---
id: LoadUser
title: Load User
sidebar_label: Load User
---

> An action that allows altering the contextual users which all the subsequent actions use. This action supports loading multiple users by providing multiple user identifiers. The last user identifier provided will also be used as the context user under which all actions execute (logged in user).

Reviewer: As a first time user, do I know what "contextual users" mean. 
Let's explain what we know
- by default, the actions refer to to the user that's currently logged in
- and that that user can be changed just for the duration of the workflow using these actions

---
id: LoadUser
title: Load User
sidebar_label: Load User
---

> An action that allows altering the user which the subsequent actions refer to. The user change affects all actions after the `Load User` action and after all actions have finished executing the user will be reverted back to the initial one. By default, all actions that need an user for their execution will refer to the currently logged in user.

Reviewer 2: ```suggestion
An action that allows altering the user which the subsequent actions refer to. The user change affects all actions after the `Load User` action and after all actions have finished executing the user will be reverted back to the initial one (eg: the logged-in user). By default, all actions that need a user for their execution will refer to the currently logged in user.
```

This action also supports loading multiple users by providing multiple user identifiers. The last user identifier provided will also be used as the context user under which all actions execute. Loading multiple users creates a hidden context entity that is leveraged by the following actions which support this:

- [Grant Role](/docs/Actions/grant-role.md)
- [Revoke Role](/docs/Actions/revoke-role.md)
- [Authorize User](/docs/Actions/authorize-user.md)
- [Unauthorize User](/docs/Actions/unauthorize-user.md)
- [Delete User](/docs/Actions/delete-user.md)
- [Send Email](/doc/Actions/send-email.md).

When used in conjunction with actions such as [Grant Role](/docs/Actions/grant-role.md), the `Grant Role` action will grant the role specified to all users loaded by this action.

## `Typical Use Cases`

> - Impersonating an user

Reviewer 2: ```suggestion
- Update another User profile
```

- Loading the user before granting the necessary roles
- Loading users before sending a mass email

## `Don't use it to`

- Load multiple users by a specific condition, use [Load Users From SQL](/docs/actions/load-users-sql.md) instead

## `Related Actions`

> 

Reviewer 2: ```suggestion
- [Update User Profile](/docs/Actions/update-user-profile.md) - updates user profile properties 
```

- [Grant Role](/docs/Actions/grant-role.md) - grants the users in the context the specified roles
- [Send Email](/doc/Actions/send-email.md) - sends an email to the users loaded in the context

## `Input Parameter Reference`

| Parameter     | Description                           | Supports Variables |
|---------------|---------------------------------------|--------------------|
| User Identifier | The identifier which will indicate which user to load into context. You can use the user ID, username or the email as identifiers. <br/> You can load multiple users by providing multiple values separated by a comma. | Yes |
| Portal ID | The identifier which will specify in which portal to look for the user indicated by the 'User Identifier'. If left blank, the action will search for the user across all portals. Valid values are either the portal ID or the portal name. | Yes |

## `Examples`

> ### `1. Load an user and grant it the Subscribers role`

Reviewer 2: ```suggestion
### `1. Load a user and grant it the Subscribers role`
```

The actions below will load the user with the identifier (user email) 'john.doe@gmail.com' into the context and then grant him the role `Subscribers` with an indefinite expiration date.

```json
{
    "Title": "Load User",
    "ActionType": "LoadUser",
    "Description": "Load the user into the context",
    "Condition": null,
    "Parameters": {
        "Id": "john.doe@gmail.com",
        "Portal": ""
    }
}
```

> 

Reviewer 2: Maybe we can add an Update User Profile Example for FirstName and LastName, Something simple
